### PR TITLE
Add --pick-most-used flag for probe --correct to automatically pick the most used version

### PR DESCRIPTION
**Pull request by spkersten**

### CODE REVIEW

**Comment by spkersten** [comment]
@alexei-sintotski Looks like the build failed because some script try to use `pub` which was deprecated and removed: `pub: command not found`

**Comment by gabrielgarciagava** [comment]
This would be a very useful feature 👍

File: test/determine_most_used_version_test.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    test("returns null of there isn't a single most used version", () {
+      final result = determineMostUsedVersion({
+        const PackageDependency.hosted(HostedPackageDependency(
+          package: 'foo',
+          version: '1.0',
+          url: '',
+          type: DependencyType.direct,
+          name: 'foo',
+        )): ['package1', 'package2'],
```
Comment (gabrielgarciagava) [inline test/determine_most_used_version_test.dart:41]: Here

File: test/determine_most_used_version_test.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        const PackageDependency.hosted(HostedPackageDependency(
+          package: 'foo',
+          version: '2.0',
+          url: '',
+          type: DependencyType.direct,
+          name: 'foo',
+        )): ['package3', 'package4'],
```
Comment (gabrielgarciagava) [inline test/determine_most_used_version_test.dart:48]: And here

**Pull request closed without merging**